libraries/botbuilder-core/botbuilder/core/teams/teams_activity_handler.py

```diff
@@ -835,3 +853,29 @@ async def on_teams_channel_restored(  # pylint: disable=unused-argument
         :returns: A task that represents the work queued to execute.
         """
         return
+
+    async def on_teams_tab_fetch(  # pylint: disable=unused-argument
+        self, turn_context: TurnContext, tab_request: TabRequest
+    ):
+        """
+        Override this in a derived class to provide logic for when a tab is fetched.
+
+        :param turn_context: A context object for this turn.
+        :param tab_request: The tab invoke request value payload.
+
+        :returns: A Tab Response for the request.
+        """
+        return
```

Comment:
I *believe* these should return a NOT_IMPLEMENTED invoke response, similar to [`on_teams_messaging_extension_fetch_task()`](https://github.com/microsoft/botbuilder-python/blob/f53312fb4344ee9501face7582c7f5e2b799a402/libraries/botbuilder-core/botbuilder/core/teams/teams_activity_handler.py#L420). [They do in .NET](https://github.com/microsoft/botbuilder-dotnet/blob/0fde312e927d7a5f651969ed133773eb38067b64/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs#L400) and I think it's only the `OnTeamDeleted/Added/etc` that don't.

Reply:
Thanks a lot for piointing this! I moved the methods to their corresponding place on added this fix